src/zenml/client.py

```diff
@@ -2236,6 +2249,69 @@ def get_run_step(self, step_run_id: UUID) -> StepRunResponseModel:
         """
         return self.zen_store.get_run_step(step_run_id)
 
+    # -------------
+    # - Artifacts -
+    # -------------
+
+    def list_artifacts(
+        self,
+        artifact_uri: Optional[str] = None,
+        artifact_store_id: Optional[UUID] = None,
+    ) -> List[ArtifactResponseModel]:
+        """Get all artifacts.
+
+        Args:
+            artifact_uri: If provided, only return artifacts with this URI.
+            artifact_store_id: If provided, only return artifacts from this
+                artifact store.
+
+        Returns:
+            A list of artifacts.
+        """
+        return self.zen_store.list_artifacts(
+            artifact_uri=artifact_uri,
+            artifact_store_id=artifact_store_id,
+        )
+
+    def get_artifact(self, artifact_id: UUID) -> ArtifactResponseModel:
+        """Get an artifact by ID.
+
+        Args:
+            artifact_id: The ID of the artifact to get.
+
+        Returns:
+            The artifact.
+        """
+        return self.zen_store.get_artifact(artifact_id)
+
+    def delete_artifact(self, artifact_id: UUID) -> None:
+        """Delete an artifact.
+
+        This will delete both the metadata of the artifact from the database
+        and the actual artifact from the artifact store.
+
+        Args:
+            artifact_id: The ID of the artifact to delete.
+        """
+        from zenml.artifact_stores.base_artifact_store import BaseArtifactStore
+        from zenml.stack.stack_component import StackComponent
+
+        artifact = self.get_artifact(artifact_id=artifact_id)
+        if artifact.artifact_store_id:
+            artifact_store_model = self.get_stack_component(
+                component_type=StackComponentType.ARTIFACT_STORE,
+                name_id_or_prefix=artifact.artifact_store_id,
+            )
+            artifact_store = StackComponent.from_model(artifact_store_model)
+            assert isinstance(artifact_store, BaseArtifactStore)
+            artifact_store.rmtree(artifact.uri)
```

Comment:
This would require you to have access to the artifact store on the client side and the requirements for the artifact store might not be installed. It would be nice to catch those errors and make it clear to the users.

Reply:
Yeah good catch!

Reply:
+1, and maybe an option to just delete the metadata?

Reply:
Added error handling and also a `--only-metadata` flag that can be passed to `zenml artifact delete`